parachain/pallets/eth-app/src/lib.rs

```diff
@@ -81,6 +92,14 @@ decl_module! {
 			let who = ensure_signed(origin)?;
 			let asset_id: BridgedAssetId = H160::zero();
 			<asset::Module<T>>::do_burn(asset_id, &who, amount)?;
+
+			let message = Message {
+				sender: who.clone(),
+				recipient: recipient,
+				amount: amount
+			};
+			T::Commitments::add(Self::address(), message.encode());
+
 			Self::deposit_event(RawEvent::Transfer(who.clone(), recipient, amount));
```

Comment:
Do we still need this?

Reply:
Yeah, still need it, see my comment above.